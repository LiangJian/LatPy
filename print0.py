def print0(rank, *args):
  if rank == 0:
    print(args)
