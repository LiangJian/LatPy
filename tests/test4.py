#from mpi4py import MPI
import timeit
import numpy as np
import cupy as cp
import time
from print0 import print0
import sys
import torch
import heat as ht

rank = ht.get_comm().rank
#comm = MPI.COMM_WORLD
#rank = comm.Get_rank()
#ngpu = comm.Get_size()

t = torch.cuda.get_device_properties(0).total_memory / 1024. / 1024. /1024. 
r = torch.cuda.memory_reserved(0) / 1024. / 1024.
a = torch.cuda.memory_allocated(0) / 1024. / 1024.
print0(rank, "Total, reserved, allocated GPU memory =", t, "GB,", r, "MB,", a, "MB,")
torch.cuda.empty_cache()
t = torch.cuda.get_device_properties(0).total_memory / 1024. / 1024. /1024. 
r = torch.cuda.memory_reserved(0) / 1024. / 1024.
a = torch.cuda.memory_allocated(0) / 1024. / 1024.
print0(rank, "Total, reserved, allocated GPU memory =", t, "GB,", r, "MB,", a, "MB,")

N = 50

size = 5000 * int(sys.argv[1])

a = ht.random.rand(size, size, dtype=ht.float64, split=0, device='gpu')
b = ht.random.rand(size, size, dtype=ht.float64, split=1, device='gpu')
t1 = time.time()
for i in range(N):
  c = a @ b
t2 = time.time()
print0(rank, "Using HeAT on gpus, matrix size {:d}, time taken is ".format(size), (t2-t1)/100., "s.")


