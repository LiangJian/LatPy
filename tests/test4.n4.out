Total, reserved, allocated GPU memory = 11.17303466796875 GB, 0.0 MB, 0.0 MB, 
Total, reserved, allocated GPU memory = 11.17303466796875 GB, 0.0 MB, 0.0 MB, 
Using HeAT on gpus, matrix size 20000, time taken is  3.2259579610824587 s. 
